[#215] Code-assist: `file` as value for responses' `type`

ghillairet opened the pull request:
This commit adds support for the core json schema, and resolves the issue with missing content assist information that is available only from that core schema. See #215

ghillairet committed:
[#215] Code-assist: `file` as value for responses' `type`

This commit adds support for the core json schema, and resolves the issue with missing content assist information that is available only from that core schema.

ghillairet committed:
Merge branch 'master' into task/content-assist

tfesenko commented on com.reprezen.swagedit/src/com/reprezen/swagedit/schema/JsonType.java:
```diff
@@ -59,6 +59,8 @@ public static JsonType valueOf(JsonNode node) {
         } else if (node.has("anyOf")) {
             return JsonType.ANY_OF;
+        } else if (node.has("allOf")) {
+            return JsonType.ALL_OF;
```
Thanks!
Can we group checks for anyOf, allOf and **oneOf** together?

tfesenko commented on com.reprezen.swagedit.tests/src/com/reprezen/swagedit/assist/SwaggerContentAssistProcessorTest.xtend:
```diff
@@ -170,18 +170,97 @@ class SwaggerContentAssistProcessorTest {
 		proposals = test.apply(processor, "2")
-//		println(proposals.map[(it as StyledCompletionProposal).replacementString])
+		values = proposals.map[(it as StyledCompletionProposal).replacementString]
+		// same without required and properties
```
Thank you for the comment, I was about to ask why are we having 28 entries here instead of 30.

tfesenko commented on com.reprezen.swagedit.tests/src/com/reprezen/swagedit/assist/SwaggerProposalProviderTest.xtend:
```diff
@@ -203,18 +203,40 @@ class SwaggerProposalProviderTest {
+			"number",
+			"object",
+			"string",
+			"file"
```
Thanks!

ghillairet committed:
[#215] Code-assist: `file` as value for responses' `type`

Make sure content assist value `null` is surrounded by quotes to avoid validation errors.
